### PR TITLE
mppp: add version 0.27 and support conan v2

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **mppp/***

- add version 0.27
- support conan v2
- remove dirty patch for boost components(it works fine on conan 1.53 at least)
- update dependencies

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by mmatisko — APPROVED** [review]

File: recipes/mppp/all/conanfile.py [hunk 1]
```diff
@@ -58,56 +64,57 @@ def requirements(self):
...
-            self.requires("boost/1.79.0")
+            self.requires("boost/1.80.0")
 
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, 11)
+            check_min_cppstd(self, self._minimum_cpp_standard)
         if self.options.with_arb:
-            raise ConanInvalidConfiguration("{}/{} doesn't supported arb (yet)".format(self.name, self.version))
+            raise ConanInvalidConfiguration(f"{self.ref} doesn't supported arb (yet)")
```
Comment (prince-chrismc) [inline recipes/mppp/all/conanfile.py:73]: Seems a little odd to have options that are not supported by the project?

The message might need some improvement... is the version too high in CCI? IS it not offered here yet?

```suggestion
            raise ConanInvalidConfiguration(f"{self.ref} doesn't support arb (yet)")
```

File: recipes/mppp/all/conanfile.py [hunk 2]
```diff
@@ -58,56 +64,57 @@ def requirements(self):
...
         if self.options.with_quadmath:
-            raise ConanInvalidConfiguration("{}/{} doesn't supported libquadmath (yet)".format(self.name, self.version))
+            raise ConanInvalidConfiguration(f"{self.ref} doesn't supported libquadmath (yet)")
```
Comment (prince-chrismc) [inline recipes/mppp/all/conanfile.py:75]: same as above

**Review by prince-chrismc — COMMENTED** [review]
Just one question about the error message for the options.

As always excellent work 🚀

File: recipes/mppp/all/conanfile.py (hunk 2 above)
Comment (toge) [reply recipes/mppp/all/conanfile.py:75]: same reason.

**Review by uilianries — DISMISSED** [review]
LGTM

**Review by prince-chrismc — COMMENTED** [review]
Can we please make the error message more descriptive? 🙏 

My non-english brain does not correctly understand whats written 🙈 trying to avoid confusing for any consumers like me 😸

**Comment by toge** [comment]
@prince-chrismc 
Thank you for your suggestion.
It is a matter of my ability to express myself in English, it's not due to yours.
Thank you for your suggestions for improvement.

**Review by uilianries — APPROVED** [review]
Arigato!

**Review by danimtb — APPROVED** [review]

**Pull request merged**